app/grandchallenge/components/forms.py

```diff
@@ -151,8 +165,60 @@ def _get_image_field(self, *, interface, current_value):
         )
 
     def _get_file_field(self, *, interface, values, current_value):
-        return self._get_default_field(
-            interface=interface, current_value=current_value
+        if current_value:
+            if isinstance(current_value, ComponentInterfaceValue):
+                # this happens on initial form load when there already is a CIV for
+                # the interface
+                return self._get_select_upload_widget_field(
+                    interface=interface,
+                    values=values,
+                    current_value=current_value,
+                )
+            else:
+                # on form submit, current_value either is the pk of an existing CIV
+                # or the UUID of a new UserUpload object
```

Comment:
You mean prepend this to `current_value` itself, or what do you mean by identifiers?

Reply:
The value of `current_value`.

Reply:
I think it would be best to pass along another parameter that identifies the widget type. I'll see if that breaks the user upload widget in other places first though

Reply:
As far as I can tell this works and doesn't break the upload widget in other places